Tensile/KernelWriterAssembly.py

```diff
@@ -1814,6 +1833,164 @@ def functionSignature(self, kernel ):
     kStr += self.comment1("DirectToLdsB=%s" % kernel["DirectToLdsB"])
     kStr += self.comment1("UseSgprForGRO=%s" % kernel["UseSgprForGRO"])
 
+    if kernel["ProblemType"]["DataType"].isHalf():
+      if kernel["ProblemType"]["HighPrecisionAccumulate"]:
+        srcSize = "2"
+        srcAlign = "2"
+        srcValueType = "F16"
+        dstSize = "2"
+        dstAlign = "2"
+        dstValueType = "F16"
+        cptSize = "4"
+        cptAlign = "4"
+        cptValueType = "F32"
+      else:
+        srcSize = "2"
+        srcAlign = "2"
+        srcValueType = "F16"
+        dstSize = "2"
+        dstAlign = "2"
+        dstValueType = "F16"
+        cptSize = "2"
+        cptAlign = "2"
+        cptValueType = "F16"
+    
+    elif kernel["ProblemType"]["DataType"].isInt8x4():
+      srcSize = "1"
+      srcAlign = "1"
+      srcValueType = "I8"
+      dstSize = "4"
+      dstAlign = "4"
+      dstValueType = "I32"
+      cptSize = "4"
+      cptAlign = "4"
+      cptValueType = "I32"
+    
+    elif kernel["ProblemType"]["DataType"].isSingle():
+      srcSize = "4"
+      srcAlign = "4"
+      srcValueType = "F32"
+      dstSize = "4"
+      dstAlign = "4"
+      dstValueType = "F32"
+      cptSize = "4"
+      cptAlign = "4"
+      cptValueType = "F32"
+    
+    elif kernel["ProblemType"]["DataType"].isDouble():
+      srcSize = "8"
+      srcAlign = "8"
+      srcValueType = "F64"
+      dstSize = "8"
+      dstAlign = "8"
+      dstValueType = "F64"
+      cptSize = "8"
+      cptAlign = "8"
+      cptValueType = "F64"
+
+    if globalParameters["CodeObjectVersion"] == "V2":
+#     Codeobject V2 metadata
+      kStr += ".amd_amdgpu_hsa_metadata\n"
+      kStr += "Version: [ 1, 0 ]\n"
+      kStr += "Kernels:\n"
+      kStr += "  - Name: %s%s" % (self.kernelName, self.endLine)
+      kStr += "    SymbolName: '%s@kd'%s" % (self.kernelName, self.endLine)
+      kStr += "    Language: OpenCL C\n"
+      kStr += "    LanguageVersion: [ 2, 0 ]\n"
+      kStr += "    Args:\n"
+
+      kStr += self.v2Argument(                               'D', dstSize, dstAlign, "GlobalBuffer", dstValueType, "Generic")
```

Comment:
For buffers, size and align should be size and align of the pointer type itself, not the pointee type. Therefore size and align for buffer should always be 8.